content/blog/2019-11-01_react-typescript-webpack-2.md

```diff
@@ -0,0 +1,193 @@
+
+
+Code splitting is very powerful but not that useful if you are just starting: worry about it when your codebase is bigger. If you
+are using React, the [documentation page on code splitting](https://reactjs.org/docs/code-splitting.html) is very well written and should
+answer most questions on how to actually use it.
+
+To use code splitting with TypeScript, you will also need to change `module` in `tsconfig.json` to `esnext`.
+
+### Analyzing bundles
+
+Once you have your bundle(s), a useful step is to actually check what they contain and whether there is fat that can be trimmed.
+[webpack-bundle-analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer) is shining for that usecase.
+
+```bash
+$ yarn add webpack-bundle-analyzer --dev
+```
+
+We only need to analyze bundles once in a while so I like to create a new script for it `"analyze": "cross-env NODE_ENV=production ANALYZE=true webpack",` and
+only instantiate the plugin when that `ANALYZE` environment variable is set to `true`. 
+Running `analyze` will give you a treemap visualisation of each package used and their size, with and without gzip.
+
+Using this tool, it becomes very easy to notice some packages taking way more spaces than they should. The most classic examples I've seen
+personally are not removing [Momentjs](https://momentjs.com/) locales for an English only site and having the full [Lodash](https://lodash.com/) while
+only using one or two functions.
+
+The commit is <https://github.com/Keats/webpack-react-typescript/commit/41190ae6e279c84c8614d5d77656b8afaed1dea3>.
+
+## Extracting CSS
+
+If you added Sass by following the previous article and looked at the bundle analyzer results, you might have noticed that the app bundle contains CSS. That's because
+we've inlined them in our configuration via `style-loader` and need to extract it to a separate CSS file in production environment instead using a plugin.
+
+```bash
+$ yarn add mini-css-extract-plugin --dev
+```
+
+The change is pretty straightforward: we load the `MiniCssExtractPlugin` loader instead of `style-loader` in production and
+add the plugin to the plugin list. The plugin can be set for every environment as it will not do anything unless the loader
+is also used.
+
+Running `build:prod` will now create a CSS file as well as a sourcemap in the `dist` folder. I'm not 100% sure the sourcemap
+is accurate as I have never used them for CSS.
+
+The commit is <https://github.com/Keats/webpack-react-typescript/commit/e97a368d67e88f2c1adb9b018b8a4164fdfd3283>.
+
+## Conclusion
+
+If you followed the articles or just cloned the repository, you should in a good place to start actually building your project.
+It might look complicated compared to [create-react-app](https://github.com/facebook/create-react-app) but this is a minimal setup
+that you understand and that only has things what you need. Well, being a JavaScript project it still pulls way too many dependencies just for that but 
+it's a good start.
```

Comment:
The conclusion could be a little more inspiring xD Kind of admitting that js is a huge pain in the ass here :P

Reply:
tu peux voir que je commençais à fatiguer :p